src/gleam_cowboy_native.erl

```diff
@@ -55,3 +56,20 @@ read_entire_body(Body, Req0) ->
         {ok, Chunk, Req1} -> {list_to_binary([Body, Chunk]), Req1};
         {more, Chunk, Req1} -> read_entire_body([Body, Chunk], Req1)
     end.
+
+parse_query_string(Query) ->
+    QS = uri_string:dissect_query(Query),
+    ProcessedQS = lists:map(
+        fun({K, V}) ->
+            {K,
+                try
+                    list_to_integer(binary_to_list(V))
```

Comment:
this will only auto-parse integers and strings but could be updated to handle other things in the future